extensions/amp-story/1.0/amp-story.js

```diff
@@ -1041,11 +1042,14 @@ export class AmpStory extends AMP.BaseElement {
 
     // Story is being prerendered: resolve the layoutCallback when the active
     // page is built. Other pages will only build if the document becomes
-    // visible.
+    // previewed or visible.
     const initialPageEl = this.element.querySelector(
       `amp-story-page#${escapeCssSelectorIdent(initialPageId)}`
     );
-    if (!this.getAmpDoc().hasBeenVisible()) {
+    if (
+      !this.getAmpDoc().hasBeenVisible() &&
+      !this.getAmpDoc().hasBeenPreviewed()
+    ) {
```

Comment:
I'm unsure whether this change is necessary

Reply:
I think it's not needed. To verify, plz check if the promise `return storyLayoutPromise;` actually resolves (i.e. if `storyLayoutPromise.then(() => console.log('hehe'));` logs when loading the page in `preview` vState, you don't need this change.